2_API_specification/Suggestions/SteffenBrinckmann/extractor_csv.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/python3
+"""extract data from .csv file
+"""
+import sys
+from io import StringIO
+import numpy as np
+import matplotlib.pyplot as plt
+
+def use(filePath, recipe='', saveFileName=None, **kwargs):
+  """
+  Args:
+    filePath (string): full path file name
+    recipe (string): supplied to guide recipes
+                     recipe is / separated hierarchical elements parent->child
+    saveFileName (string): if given, save the image to this file-name
+
+  Returns:
+    dict: containing image, metaVendor, metaUser, recipe
+  """
+  # Extractor for fancy instrument
+  data = np.loadtxt(filePath, delimiter=',')
+  if recipe.endswith('red'):              #: Draw with red curve
+    plt.plot(data[:,0], data[:,1],'r')
+  else:                                   #: Default | blueish curve
+    plt.plot(data[:,0], data[:,1])
+  metaUser = {'max':data[:,1].max(), 'min':data[:,1].min()}
+  recipe = 'csv'
+
+  #save to file
+  if saveFileName is not None:
+    plt.savefig(saveFileName, dpi=150, bbox_inches='tight')
+
+  #convert axes to svg image
+  figfile = StringIO()
+  plt.savefig(figfile, format='svg')
+  image = figfile.getvalue()
+
+  # return everything
+  return {'image':image, 'recipe':recipe, 'metaVendor':{}, 'metaUser':metaUser}
```

Comment:
Might be worth thinking if pydantic might be useful to give some additional features (e.g. validation) and mapping into other formats/classes

Reply:
:-) It goes again into the competition with the entry-barrier. Pydantic is great and I read on it some time ago and it might have great features and be the thing to use here. But one needs to know it or there has to easy to understand examples that it becomes easy. I did not find the time to look into it, and so, yes ... it might be great.